Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAppWipeRequest/MSFT_IntuneAppWipeRequest.psm1

```diff
@@ -0,0 +1,592 @@
+            Version               = $instance.version
+            PatchVersion          = $instance.patchVersion
+            AppIdentifier         = $instance.appIdentifier.packageId
+            OperationsContext     = $instance."operations@odata.context"
+            Operations            = $instance.operations
+        }
+        return [System.Collections.Hashtable] $results
+    }
+    catch
+    {
+        Write-Verbose -Message $_
+        New-M365DSCLogEntry -Message 'Error retrieving data:' `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return $nullResult
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Wipe,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens,
+
+        [Parameter()]
+        [System.String]
+        $deviceTag,
+
+        [Parameter()]
+        [System.String]
+        $deviceName,
+
+        [Parameter()]
+        [System.String]
+        $version,
+
+        [Parameter()]
+        [System.String]
+        $patchVersion,
+
+        [Parameter()]
+        [System.Collections.Hashtable]
+        $appIdentifier
+    )
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName.Replace('MSFT_', '')
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $currentInstance = Get-TargetResource @PSBoundParameters
+
+    $setParameters = Remove-M365DSCAuthenticationParameter -BoundParameters $PSBoundParameters
+
+    # CREATE
+    if ($Ensure -eq 'Present' -and $currentInstance.Ensure -eq 'Absent')
+    {
+        New-MgBetaDeviceAppManagementTargetedManagedAppConfiguration -Id $Wipe `
+        -DisplayName $deviceName `
+        -Description $deviceTag `
+        -ApplicationVersion $applicationVersion `
+        -ManagementSdkVersion $managementSdkVersion `
+        -PlatformVersion $platformVersion `
+        -PatchVersion $patchVersion `
+        @SetParameters
```

Comment:
Wipe is already passed as the Id, shouldn't it be removed from the $SetParameters? Feels like this is going to throw an error.